server/datastore/mysql/migrations/tables/20230721161508_QueriesDataMigrator.go

```diff
@@ -0,0 +1,458 @@
+				INNER JOIN packs p ON sq.pack_id = p.id
+		WHERE p.pack_type <> 'global'
+		AND p.pack_type IS NOT NULL`
+
+	rows, err := tx.Query(selectStmt)
+	if err != nil {
+		return fmt.Errorf("error executing 'Query' for scheduled queries from team packs: %s", err)
+	}
+	defer rows.Close()
+
+	var args []interface{}
+	var nRows int
+
+	for rows.Next() {
+		nRows += 1
+		var query QueryWithScheduledFields
+		if err := rows.Scan(
+			&query.Name,
+			&query.Description,
+			&query.Query.Query,
+			&query.AuthorID,
+			&query.Saved,
+			&query.ObserverCanRun,
+			&query.ScheduledQID,
+			&query.ScheduledQueryInterval,
+			&query.ScheduledQSnapshot,
+			&query.ScheduledQRemoved,
+			&query.ScheduledQueryPlatform,
+			&query.ScheduledQueryVersion,
+			&query.PackType,
+		); err != nil {
+			return fmt.Errorf("error executing 'Scan' for scheduled queries from team packs: %s", err)
+		}
+
+		teamIDParts := strings.Split(query.PackType, "-")
+		teamID, err := strconv.Atoi(teamIDParts[1])
+		if err != nil {
+			return fmt.Errorf("error parsing TeamID for scheduled queries from team packs: %s", err)
+		}
+
+		var loggingType string
+		if query.ScheduledQSnapshot != nil && *query.ScheduledQSnapshot {
+			loggingType = "snapshot"
+		}
+		if loggingType == "" && query.ScheduledQRemoved != nil {
+			if *query.ScheduledQRemoved {
+				loggingType = "differential"
+			} else {
+				loggingType = "differential_ignore_removals"
+			}
+		}
+
+		args = append(args,
+			teamID,
+			teamIDParts[1],
+			fmt.Sprintf("%s - %d", query.Name, query.ScheduledQID),
+			query.Description,
+			query.Query.Query,
+			query.AuthorID,
+			query.Saved,
+			query.ObserverCanRun,
+			query.ScheduledQueryPlatform,
+			query.ScheduledQueryVersion,
+			query.ScheduledQueryInterval,
+			loggingType,
+			true,
+		)
+	}
+	if err := rows.Err(); err != nil {
+		return fmt.Errorf("error on 'rows' for scheduled queries from team packs: %s", err)
+	}
+	if err := rows.Close(); err != nil {
+		return fmt.Errorf("error closing reader from team packs: %s", err)
+	}
+
+	if len(args) == 0 {
+		return nil
+	}
+
+	insertStmt := `
+		INSERT INTO queries (
+			team_id,
+			team_id_char,
+			name,
+			description,
+			query,
+			author_id,
+			saved,
+			observer_can_run,
+			platform,
+			min_osquery_version,
+			schedule_interval,
+			logging_type,
+			automations_enabled 
+		) VALUES %s`
+
+	placeHolders := strings.TrimSuffix(strings.Repeat("( ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ? ),", nRows), ",")
+	if _, err = tx.Exec(fmt.Sprintf(insertStmt, placeHolders), args...); err != nil {
+		return fmt.Errorf("error executing 'Exec' for scheduled queries from team packs: %s", err)
+	}
+
+	return nil
+}
+
+func _20230719152138_migrate_non_scheduled(tx *sql.Tx) error {
+	// If the query is not scheduled, then it stays global except if it was created by a team user,
+	// in which case the query is duplicated as a team query iff the user is a mantainer of the team.
```

Comment:
I think this is more accurate? :

```suggestion
	// in which case the query is duplicated as a team query iff the user is an admin or maintainer of the team.
```

Reply:
Done in https://github.com/fleetdm/fleet/pull/12855/commits/cdfebb29280de6267ebb534320e9eb8f0a52b36d